### ISSUE TITLE
Decide how elementwise function applications should be printed

### ISSUE THREAD

**Issue by Upabjojr** [sympy/sympy#16680]
The notation `sin(X...)` for the elementwise function application of a function on a matrix has been introduced in https://github.com/sympy/sympy/pull/16675 .

`X.applyfunc(sin)` prints as `sin(X...)`.

It is distinguished from `sin(X)` which is the function applied to the matrix (i.e. convergence of Taylor series).

This feature is new to the master branch and can be freely changed before SymPy 1.5 release.

Should we find a better way to express this application?

**Comment by cbm755** [comment]
I'm not a fan of `sin(A...)`, looks too much like some continuing iteration.  If we cannot come up with something better maybe we should consider reverting this change?

Here's one suggestion for discussion: pprint unicode to `sin∘(A)` and pprint ascii to `sin.(A)`.  Not necessarily "+1" for this, just throwing it out there...

My rational is that Julia uses `sin.(A)` to execute this operation, presumably inspired by the common Hadamard product ascii notation `.*` (e.g., Matlab/Octave/Julia).

**Comment by Upabjojr** [comment]
>  pprint unicode to `sin∘(A)` and pprint ascii to `sin.(A)`

Very nice idea.

What about creating some other shortcuts? For example `sin.elementwise(A)` instead of `A.applyfunc(sin)`.

Or maybe the recently introduced matrix product operator `@`. For example: `sin@A`.

**Comment by Upabjojr** [comment]
I mean, `scalar@matrix` already behaves like this:
```python
In [1]: var("x")                                                                                                           
Out[1]: x

In [2]: A = MatrixSymbol("A", 3, 3)                                                                                        

In [3]: (x@A).as_explicit()                                                                                                
Out[3]: 
⎡x⋅A₀₀  x⋅A₀₁  x⋅A₀₂⎤
⎢                   ⎥
⎢x⋅A₁₀  x⋅A₁₁  x⋅A₁₂⎥
⎢                   ⎥
⎣x⋅A₂₀  x⋅A₂₁  x⋅A₂₂⎦
```

You could replace `x` by a function and apply it to every element.

**Comment by cbm755** [comment]
That sounds like a discussion for another issue

**Comment by oscarbenjamin** [comment]
@Upabjojr is this now fixed by #17416?

**Comment by Upabjojr** [comment]
Yes, unless you want to suggest an alternative way of printing.

**Comment by oscarbenjamin** [comment]
In my terminal the "operator low ring" character looks like a normal dot
```
In [3]: X.applyfunc(sin)                                                                                                                                      
Out[3]: sin˳(X)
```
but I think that's not an issue for sympy.

The repr output is still unchanged after #17416 though:
```
In [17]: X.applyfunc(sin)                                                                                                                      
Out[17]: sin˳(X)

In [18]: repr(_)                                                                                                                               
Out[18]: 'sin(X...)'
```

**Comment by Upabjojr** [comment]
I have pushed the `repr` correction in someone else's PR, which has probably not been merged.

Don't remember which PR it is.

**Comment by oscarbenjamin** [comment]
@Upabjojr the 1.5 release is coming soon. Can the repr be fixed?

**Comment by oscarbenjamin** [comment]
Closing as fixed in #17929 for 1.5
